### ISSUE TITLE
Callback is called if you call runWithPremissions twice

### ISSUE THREAD

**Issue by mariciv** [QuickPermissions/QuickPermissions-Kotlin#5]
Hi there. I ran into an issue.

After revoking the permission from app settings screen I can't get to the system permission dialog anymore. The lib automatically calls callback function even though permission is not allowed. 

Here is the code: 
```
runWithPermissions(permission,
                        options = QuickPermissionsOptions(
                            handlePermanentlyDenied = true,
                            handleRationale = true,
                            permanentDeniedMethod = { denyFunction.invoke() },
                            permissionsDeniedMethod = { denyFunction.invoke() },
                            rationaleMethod = {
                                // We call this since on permission denied is not called
                                denyFunction.invoke()
                            }),
                        callback = { Timber.d("Permission callback") })
```

**Comment by kirtan403** [comment]
Hi @mariciv . So your issue is resolved now? Or you are still facing some unexpected behavior?

**Comment by mariciv** [comment]
@kirtan403 I was able to resolve my issue, but I think library could handle this case a bit better. 
So if `runWithPermissions` method is called twice, before user denies or allows permission in the system dialog, action callback shouldn't be called.

**Comment by kirtan403** [comment]
Thanks, @mariciv . Actually it works on the singleton objects for the requests and response. That's why that happened. I'll add this to the enhancement. Thanks for making things clear 😄

**Comment by kirtan403** [comment]
Fixed (Ref: 62f54a41)

**Comment by kirtan403** [comment]
New version released with fix: https://github.com/QuickPermissions/QuickPermissions-Kotlin/releases/tag/0.4.0
